nncf/experimental/onnx/algorithms/quantization/min_max_quantization.py

```diff
@@ -103,52 +102,51 @@ def _get_quantizer_setup(self, model: onnx.ModelProto):
         final_setup = solver.get_final_quantizer_setup(finalized_proposal)
         return final_setup
 
+    def _append_weight_quantization_target_point(self, quantization_point) -> None:
+        weight_quantization_target_point = ONNXTargetPoint(TargetType.OPERATION_WITH_WEIGHTS,
+                                                           quantization_point.insertion_point.target_node_name)
+        # It prevents the duplicate weight quantizers from being added.
+        # It can happen when you have layers that share the identical weight tensor.
+        if weight_quantization_target_point in self._quantization_target_points:
```

Comment:
This condition should be checked outside the function body. Otherwise the function name does not correspond to the function effect. It does not always "append" weight quantization target point. One of the solutions could be to add "maybe" to the function name, but see if you can refactor to have the functions do their thing without any "maybes" attached.

Reply:
What If I make `add` instead `append` and transfrom the `_quantization_target_points` from `List` to `Set`

Reply:
That would be a good solution, provided that the target points are properly hashable.

Reply:
Done